daemon/Libraries/Utilities/Architecture/ArchitectureUtility.cs

```diff
@@ -0,0 +1,77 @@
+using System;
+using System.IO;
+using Medallion.Shell;
+using Spectero.daemon.Libraries.Config;
+using Spectero.daemon.Libraries.Core.ProcessRunner;
+
+namespace Spectero.daemon.Libraries.Utilities.Architecture
+{
+    public class ArchitectureUtility : IArchitectureUtility
+    {
+        private readonly IProcessRunner _processRunner;
+
+        public ArchitectureUtility(IProcessRunner processRunner)
+        {
+            _processRunner = processRunner;
+        }
+
+        public string GetArchitecture()
+        {
+            if (AppConfig.isWindows) return GetWindowsArchitecture();
+            if (AppConfig.isLinux) return GetLinuxArchitecture();
+            if (AppConfig.isUnix) return GetUnixArchitecture();
+            return null;
```

Comment:
Don't return null, that's ambiguous behavior (unless logging has been added elsewhere to specifically detect (and react to) this null.

Silent failures are not nice.

Reply:
What would you suggest I do then, as this is returning a type of string?

Reply:
That depends on what exactly this is being used for. The way I see it, you should make it throw an InternalError, and then catch it on the UpdaterJob.

On being caught, you should disable the updater (because update is not possible as the arch is completely unknown to us).

Reply:
I'd like to perform the latter if that's okay with you, but at the same time it means there's an edgecase that we may have to find support for.

Reply:
Implemented.